cobigen-plugins/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/test/java/com/devonfw/cobigen/openapiplugin/unittest/matcher/OpenAPIMatcherTest.java

```diff
@@ -73,4 +78,74 @@ public void testInvalidComponentDefMatching() {
 
     assertThat(matches).isFalse();
   }
+
+  /**
+   * Test if the generation is successful and without warnings, if a requested variable is assigned
+   */
+  @Test
+  public void testXRootPackageVariableAssigned() {
+
+    ComponentDef componentDef = new ComponentDef();
+    componentDef.setName("Tablemanagement");
+    componentDef.setUserProperty("x-rootpackage", "com.devonfw");
+
+    OpenAPIMatcher matcher = new OpenAPIMatcher();
+    GenerationReportTo report = new GenerationReportTo();
+    List<VariableAssignmentTo> va = new ArrayList<>();
+    va.add(new VariableAssignmentTo("extension", "rootPackage", "x-rootpackage", false));
+
+    matcher.resolveVariables(new MatcherTo("element", "ComponentDef", componentDef), va, report);
+    assertThat(report.getWarnings().size()).isEqualTo(0);
+  }
+
+  /**
+   * Test if the generation is successful and the report contains warnings, if a requested but not mandatory variable
+   * isn't given.
+   */
+  @Test
+  public void testMissingXRootPackageVariableNotMandatory() {
+
+    ComponentDef componentDef = new ComponentDef();
+    componentDef.setName("Tablemanagement");
+
+    OpenAPIMatcher matcher = new OpenAPIMatcher();
+    GenerationReportTo report = new GenerationReportTo();
+    List<VariableAssignmentTo> vaOptionalXRootPackage = new ArrayList<>();
+    vaOptionalXRootPackage.add(new VariableAssignmentTo("extension", "rootPackage", "x-rootpackage", false));
+
+    matcher.resolveVariables(new MatcherTo("element", "ComponentDef", componentDef), vaOptionalXRootPackage, report);
+    assertThat(report.getWarnings().get(0))
+        .containsSequence("The property x-rootpackage was requested in a variable assignment "
+            + "although the input does not provide this property. Setting it to empty");
+
+    List<VariableAssignmentTo> vaMandatoryXRootPackage = new ArrayList<>();
+    vaMandatoryXRootPackage.add(new VariableAssignmentTo("extension", "rootPackage", "x-rootpackage", true));
+
+    matcher.resolveVariables(new MatcherTo("element", "ComponentDef", componentDef), vaMandatoryXRootPackage, report);
+    assertThat(report.getErrors().get(0).getMessage())
+        .containsSequence("The property x-rootpackage was required in a variable assignment "
+            + "although the input does not provide this property. "
+            + "Please add the required attribute in your input file or set the \"mandatory\" attribute to \"false\". ");
```

Comment:
I feel we should externalize this string and reuse it By making use of string formatter. But the string already occurred quite some time and would make small changes nasty to be done

Reply:
Pulled the error and warning message into a method in the Constants class, passing the value that was/wan't set to mandatory as a parameter